src/ucp/rma/flush.c

```diff
@@ -476,14 +509,25 @@ static unsigned ucp_worker_flush_progress(void *arg)
         }
     }
 
-    if ((worker->context->config.ext.flush_worker_eps) &&
+    if (worker->context->config.ext.flush_worker_eps &&
         (&next_ep->ep_list != &worker->all_eps)) {
-        /* Some endpoints are not flushed yet. Take next endpoint from the list
-         * and start flush operation on it.
-         */
-        ep                        = ucp_ep_from_ext_gen(next_ep);
-        req->flush_worker.next_ep = ucs_list_next(&next_ep->ep_list,
-                                                  ucp_ep_ext_gen_t, ep_list);
+        /* Some endpoints are not flushed yet. Take the endpoint from the list
+         * and start flush operation on it. */
+        ep = ucp_ep_from_ext_gen(next_ep);
+
+        ucp_worker_flush_req_set_next_ep(req, ucs_list_next(&next_ep->ep_list,
```

Comment:
pass `&next_ep->ep_list` as param to ucp_worker_flush_req_set_next_ep: replace "ep" param by "ep_iter"